src/main/java/uk/gov/ons/ctp/response/sample/ingest/CsvIngesterCensus.java

```diff
@@ -83,7 +83,7 @@ public CsvIngesterCensus() {
     columnPositionMappingStrategy.setColumnMapping(COLUMNS);
   }
 
-  public SampleSummary ingest(MultipartFile file)
+  public SampleSummary ingest(MultipartFile file, String collectionExerciseId)
       throws Exception {
 
     CSVReader csvReader = new CSVReader(new InputStreamReader(file.getInputStream()), ':');
```

Comment:
You are not closing the reader. Or you put it in a try with resource , CsvReader implement closabel interface. Ir you have to explicitely close it. I would go to for a try finally in case.